spyder/widgets/panels/scrollflag.py

```diff
@@ -34,98 +56,154 @@ def paintEvent(self, event):
+            painter.setPen(pen_color)
+            brush_color = QColor(Qt.gray)
+            brush_color.setAlphaF(.5)
+            painter.setBrush(QBrush(brush_color))
+            painter.drawRect(self.make_slider_range(cursor_pos))
+
+    def enterEvent(self, event):
+        """Override Qt method"""
+        self.update()
+
+    def leaveEvent(self, event):
+        """Override Qt method"""
+        self.update()
+
+    def mouseMoveEvent(self, event):
+        """Override Qt method"""
+        self.update()
 
     def mousePressEvent(self, event):
         """Override Qt method"""
         vsb = self.editor.verticalScrollBar()
-        value = self.position_to_value(event.pos().y()-1)
-        vsb.setValue(value-.5*vsb.pageStep())
+        value = self.position_to_value(event.pos().y())
+        vsb.setValue(value-vsb.pageStep()/2)
 
-    def get_scale_factor(self, slider=False):
+    def get_scrollbar_position_height(self):
+        """Return the pixel span height of the scrollbar area in which
+        the slider handle may move"""
+        vsb = self.editor.verticalScrollBar()
+        style = vsb.style()
+        opt = QStyleOptionSlider()
+        vsb.initStyleOption(opt)
+
+        # Get the area in which the slider handle may move.
+        groove_rect = style.subControlRect(
+                QStyle.CC_ScrollBar, opt, QStyle.SC_ScrollBarGroove, self)
+
+        return float(groove_rect.height())
+
+    def get_scrollbar_value_height(self):
+        """Return the value span height of the scrollbar"""
+        vsb = self.editor.verticalScrollBar()
+        return vsb.maximum()-vsb.minimum()+vsb.pageStep()
+
+    def get_scale_factor(self):
         """Return scrollbar's scale factor:
         ratio between pixel span height and value span height"""
-        delta = 0 if slider else 2
-        vsb = self.editor.verticalScrollBar()
-        position_height = vsb.height()-delta-1
-        value_height = vsb.maximum()-vsb.minimum()+vsb.pageStep()
-        return float(position_height)/value_height
+        return (self.get_scrollbar_position_height() /
+                self.get_scrollbar_value_height())
 
-    def value_to_position(self, y, slider=False):
-        """Convert value to position"""
-        offset = 0 if slider else 1
+    def value_to_position(self, y):
+        """Convert value to position in pixels"""
         vsb = self.editor.verticalScrollBar()
-        return (y-vsb.minimum())*self.get_scale_factor(slider)+offset
+        return (y-vsb.minimum())*self.get_scale_factor()+self.offset
 
-    def position_to_value(self, y, slider=False):
-        """Convert position to value"""
-        offset = 0 if slider else 1
+    def position_to_value(self, y):
+        """Convert position in pixels to value"""
         vsb = self.editor.verticalScrollBar()
-        return vsb.minimum()+max([0, (y-offset)/self.get_scale_factor(slider)])
+        return vsb.minimum()+max([0, (y-self.offset)/self.get_scale_factor()])
```

Comment:
This is causing an zero division error in shorter files (when the scrollbar isn't displayed) when clicking over the scrollflagarea

```
Traceback (most recent call last):
  File "/home/rlaverde/code/spyder/spyder/widgets/panels/scrollflag.py", line 127, in mousePressEvent
    value = self.position_to_value(event.pos().y())
  File "/home/rlaverde/code/spyder/spyder/widgets/panels/scrollflag.py", line 163, in position_to_value
    return vsb.minimum()+max([0, (y-self.offset)/self.get_scale_factor()])
ZeroDivisionError: float division by zero
```

Reply:
Nice catch! thanks.